EventFilter/CTPPSRawToDigi/plugins/CTPPSPixelRawToDigi.cc

```diff
@@ -0,0 +1,90 @@
+/*
+ * adapting to CTPPS pixel detector March 2017 - F.Ferro
+ */
+
+
+#include "EventFilter/CTPPSRawToDigi/interface/CTPPSPixelRawToDigi.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+#include "DataFormats/DetId/interface/DetIdCollection.h"
+
+#include "EventFilter/CTPPSRawToDigi/interface/CTPPSPixelDataFormatter.h"
+
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+using namespace std;
+
+
+CTPPSPixelRawToDigi::CTPPSPixelRawToDigi( const edm::ParameterSet& conf ) 
+  : config_(conf)
+
+{
+
+  tFEDRawDataCollection = consumes <FEDRawDataCollection> (config_.getParameter<edm::InputTag>("InputLabel"));
+
+// Products
+  produces< edm::DetSetVector<CTPPSPixelDigi> >();
+
+//CablingMap could have a label //Tav
+  mappingLabel = config_.getParameter<std::string> ("mappingLabel"); //RPix
+
+}
+
+
+CTPPSPixelRawToDigi::~CTPPSPixelRawToDigi() {
+  edm::LogInfo("CTPPSPixelRawToDigi")  << " CTPPSPixelRawToDigi destructor!";
+
+}
+
+void CTPPSPixelRawToDigi::produce( edm::Event& ev,
+				   const edm::EventSetup& es) 
+{
+
+  edm::ESHandle<CTPPSPixelDAQMapping> mapping;
+  es.get<CTPPSPixelDAQMappingRcd>().get(mappingLabel, mapping);
+
+
+
+  fedIds   = mapping->fedIds();
+
+  edm::Handle<FEDRawDataCollection> buffers;
+  ev.getByToken(tFEDRawDataCollection, buffers);
+
+// create product (digis & errors)
+  auto collection = std::make_unique<edm::DetSetVector<CTPPSPixelDigi>>();
+
+//    HERE PUT OUR OWN FORMATTER
+
+  CTPPSPixelDataFormatter formatter(mapping->ROCMapping);
+
+  bool errorsInEvent = false; 
+  for (auto aFed = fedIds.begin(); aFed != fedIds.end(); ++aFed) {
+    int fedId = *aFed;
+//    int FMC = 0;
+//cout << "FEDID:   " << fedId << endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant